### PR TITLE
stock_reserve: port to 9.0

### PR DESCRIPTION
**Pull request by katyukha**
_stock_reserve_ only modifications from PRs #186 #187 #188

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ addon | version | summary
 [stock_available_immediately](stock_available_immediately/) | 9.0.1.1.0 | Ignore planned receptions in quantity available to promise
 [stock_available_mrp](stock_available_mrp/) | 9.0.1.0.0 | Consider the production potential is available to promise
 [stock_partner_lot](stock_partner_lot/) | 9.0.1.0.0 | Show lots on the partners that own them
+[stock_reserve](stock_reserve/) | 9.0.0.2.0 | Stock reservations on products
```
Comment (lasley) [inline README.md]: This file is maintained automatically & editing it as such will cause conflicts with our automation. Please reverse change

File: stock_reserve/__openerp__.py [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 {'name': 'Stock Reservation',
  'summary': 'Stock reservations on products',
- 'version': '8.0.0.2.0',
+ 'version': '9.0.0.2.0',
```
Comment (lasley) [inline stock_reserve/__openerp__.py]: 9.0.1.0.0

File: stock_reserve/model/stock_reserve.py [hunk 3]
```diff
@@ -96,35 +126,26 @@ def _default_picking_type_id(self):
...
         'picking_type_id': _default_picking_type_id,
-        'location_id': _default_location_id,
         'location_dest_id': _default_location_dest_id,
         'product_uom_qty': 1.0,
     }
 
     @api.multi
     def reserve(self):
-        """ Confirm a reservation
+        """ Confirm a reservations
```
Comment (lasley) [inline stock_reserve/model/stock_reserve.py]: s/Confirm a reservations/Confirm reservations

File: stock_reserve/model/stock_reserve.py [hunk 4]
```diff
@@ -96,35 +126,26 @@ def _default_picking_type_id(self):
...
 
         The reservation is done using the default UOM of the product.
         A date until which the product is reserved can be specified.
         """
-        self.date_expected = fields.Datetime.now()
-        self.move_id.action_confirm()
-        self.move_id.picking_id.action_assign()
+        self.write({'date_expected': fields.Datetime.now()})
+        self.mapped('move_id').action_confirm()
```
Comment (lasley) [inline stock_reserve/model/stock_reserve.py:147]: Why mapped on m2o?

File: stock_reserve/model/stock_reserve.py [hunk 5]
```diff
@@ -96,35 +126,26 @@ def _default_picking_type_id(self):
...
+        self.mapped('move_id.picking_id').action_assign()
```
Comment (lasley) [inline stock_reserve/model/stock_reserve.py]: Why mapped on m2o?

**Review by lasley — CHANGES_REQUESTED** [review]
Thanks for the submission @katyukha - comments inline

File: stock_reserve/model/stock_reserve.py (hunk 4 above)
Comment (pedrobaeza) [reply stock_reserve/model/stock_reserve.py:147]: `self` can contain more than one record, so mapped is needed

**Comment by katyukha** [comment]
Fixes sent) Thanks for review)

**Review by lasley — APPROVED** [review]

**Comment by ghost** [comment]
:+1:

**Pull request merged**